src/main/java/uk/gov/hmcts/ccd/data/caseaccess/AMCaseUserRepository.java

```diff
@@ -1,39 +1,99 @@
+import uk.gov.hmcts.reform.amlib.enums.Permission;
+import uk.gov.hmcts.reform.amlib.models.ExplicitAccessGrant;
+import uk.gov.hmcts.reform.amlib.models.ExplicitAccessMetadata;
+import uk.gov.hmcts.reform.amlib.models.ResourceDefinition;
+import uk.gov.hmcts.reform.amlib.models.UserCaseRolesEnvelope;
+import uk.gov.hmcts.reform.amlib.models.UserCasesEnvelope;
 
 import javax.inject.Named;
 import javax.inject.Singleton;
-import javax.persistence.EntityManager;
-import javax.persistence.PersistenceContext;
+import javax.sql.DataSource;
+import javax.transaction.Transactional;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.amlib.enums.Permission.READ;
 
 @Named
 @Singleton
-@Qualifier(AMCaseUserRepository.QUALIFIER)
+@Qualifier(AMCaseUserRepository.ACCESS_MANAGEMENT_QUALIFIER)
 public class AMCaseUserRepository implements CaseUserRepository {
 
-    public static final String QUALIFIER = "am";
+    public static final String ACCESS_MANAGEMENT_QUALIFIER = "accessManagement";
+    private static final String CASE_CONSTANT = "case";
+
+    private AccessManagementService accessManagementService;
 
-    @PersistenceContext
-    private EntityManager em;
+    public AMCaseUserRepository(@Qualifier("amDataSource") DataSource dataSource) {
+        accessManagementService = new AccessManagementService(dataSource);
+    }
 
     @Override
+    @Transactional
     public void grantAccess(String jurisdictionId, String caseTypeId, String caseReference, Long caseId, String userId, String caseRole) {
+        ResourceDefinition resourceDefinition =
+            //TODO: What should be the resourceType and resourceName.
+            new ResourceDefinition(jurisdictionId, CASE_CONSTANT, caseReference);
+
+        ExplicitAccessGrant explicitAccessGrant = ExplicitAccessGrant.builder()
+            .accessorType(AccessorType.USER)
+            .accessorIds(Collections.singleton(userId))
+            .resourceId(caseId.toString())
+            .attributePermissions(getAttributePermissions())
+            .resourceDefinition(resourceDefinition)
+            .relationship(caseRole)
+            .build();
+
+        accessManagementService.grantExplicitResourceAccess(explicitAccessGrant);
+    }
+
+    private Map<JsonPointer, Set<Permission>> getAttributePermissions() {
+        Map<JsonPointer, Set<Permission>> attributePermissionMap = new HashMap<JsonPointer, Set<Permission>>();
+            //TODO: What should be the permission set? Just read or CRUD?
+        attributePermissionMap.put(JsonPointer.valueOf(""), ImmutableSet.of(READ));
+        return attributePermissionMap;
     }
 
     @Override
+    @Transactional
     public void revokeAccess(String jurisdictionId, String caseTypeId, String caseReference, Long caseId, String userId, String caseRole) {
+        ExplicitAccessMetadata explicitAccessMetadata =
```

Comment:
Should not we call it `explicitAccessRevoke` to be consistent with the above `explicitAccessGrant`?

Reply:
This is a POJO class and the change will be required on Access Management codebase. I will talk to the AM team on this.